bibtexparser/bibdatabase.py

```diff
@@ -103,6 +116,48 @@ def expand_string(self, name):
         except KeyError:
             raise(UndefinedString(name))
 
+    def _add_missing_from_crossref_entry(self, entry, dependencies=set()):
+        if entry['ID'] in self._crossref_updated:
+            return
+
+        if entry['_crossref'] not in self._entries_dict:
+            logger.error("Crossref reference %s for %s is missing.",
+                         entry['_crossref'],
+                         entry['ID'])
+            return
+
+        if entry['_crossref'] in dependencies:
+            logger.error("Circular crossref dependency: %s->%s->%s.",
+                         "->".join(dependencies),
+                         entry['ID'],
+                         entry['_crossref'])
+            return
+
+        crossref_entry = self._entries_dict[entry['_crossref']]
+        if '_crossref' in crossref_entry:
+            dependencies.add(entry['ID'])
+            self._add_missing_from_crossref_entry(crossref_entry, dependencies)
+            dependencies.remove(entry['ID'])
+
+        from_crossref = {bibfield: bibvalue
+                         for (bibfield, bibvalue) in crossref_entry.items()
+                         if bibfield not in entry.keys() and
+                            bibfield not in self.not_update_by_crossref}
+
+        for bibfield, bibvalue in from_crossref.items():
+            entry[bibfield] = bibvalue
```

Comment:
`entry.update(from_crossref)`

Reply:
Fixed in e0ab2c257bfb9853855b955b845a14aa7d7ee7d3.